packages/react/src/components/Autofill/Autofill.tsx

```diff
@@ -326,12 +326,16 @@ export class Autofill extends React.Component<IAutofillProps, IAutofillState> im
     }
 
     // eslint-disable-next-line deprecation/deprecation
-    const { onInputChange, onInputValueChange } = this.props;
+    const { onInputChange, onInputValueChange, updateValueInWillReceiveProps } = this.props;
     if (onInputChange) {
       newValue = onInputChange?.(newValue, composing) || '';
     }
 
-    this.setState({ inputValue: newValue }, () => onInputValueChange?.(newValue, composing));
+    // if value is controlled, then we should not update the value in state
```

Comment:
minor nit: this is only when controlled through `updateValueInWillReceiveProps`, and not when controlled through `props.value`

```suggestion
    // if value is controlled in updateValueInWillReceiveProps, then we should not update the value in state now
```